File: frontend/src/main/java/com/innertrack/model/ChatLock.java
package com.innertrack.model;

import java.time.LocalDateTime;

public class ChatLock {
    private int id;
    private int userId;
    private String reason;
    private LocalDateTime lockedAt;
    private LocalDateTime lockedUntil; // null = permanent
    private int lockedBy;
    private boolean active;

    // Transient
    private String userName;

    public ChatLock() {
    }

    public ChatLock(int userId, String reason, LocalDateTime lockedUntil, int lockedBy) {
        this.userId = userId;
        this.reason = reason;
        this.lockedUntil = lockedUntil;
        this.lockedBy = lockedBy;
        this.active = true;
    }

    public boolean isExpired() {
        return lockedUntil != null && lockedUntil.isBefore(LocalDateTime.now());
    }

    public boolean isPermanent() {
        return lockedUntil == null;
    }

    // ── Getters ──
    public int getId() {
        return id;
    }

    public int getUserId() {
        return userId;
    }

    public String getReason() {
        return reason;
    }

    public LocalDateTime getLockedAt() {
        return lockedAt;
    }

    public LocalDateTime getLockedUntil() {
        return lockedUntil;
    }

    public int getLockedBy() {
        return lockedBy;
    }

    public boolean isActive() {
        return active;
    }

    public String getUserName() {
        return userName;
    }

    // ── Setters ──
    public void setId(int id) {
        this.id = id;
    }

    public void setUserId(int userId) {
        this.userId = userId;
    }

    public void setReason(String reason) {
        this.reason = reason;
    }

    public void setLockedAt(LocalDateTime lockedAt) {
        this.lockedAt = lockedAt;
    }

    public void setLockedUntil(LocalDateTime lockedUntil) {
        this.lockedUntil = lockedUntil;
    }

    public void setLockedBy(int lockedBy) {
        this.lockedBy = lockedBy;
    }

    public void setActive(boolean active) {
        this.active = active;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }
}

File: backend/src/Entity/ChatLock.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: 'chat_lock')]
class ChatLock
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(name: 'user_id', referencedColumnName: 'id', nullable: false)]
    private User $user;

    #[ORM\Column(length: 255)]
    private string $reason;

    #[ORM\Column(name: 'locked_at', type: 'datetime')]
    private \DateTimeInterface $lockedAt;

    #[ORM\Column(name: 'locked_until', type: 'datetime', nullable: true)]
    private ?\DateTimeInterface $lockedUntil = null;

    #[ORM\Column(name: 'locked_by')]
    private int $lockedBy;

    #[ORM\Column(name: 'is_active', type: 'boolean', options: ['default' => true])]
    private bool $isActive = true;

    public function __construct()
    {
        $this->lockedAt = new \DateTime();
    }

    public function getId(): ?int { return $this->id; }
    public function getUser(): User { return $this->user; }
    public function setUser(User $user): self { $this->user = $user; return $this; }
    public function getReason(): string { return $this->reason; }
    public function setReason(string $reason): self { $this->reason = $reason; return $this; }
    public function getLockedAt(): \DateTimeInterface { return $this->lockedAt; }
    public function getLockedUntil(): ?\DateTimeInterface { return $this->lockedUntil; }
    public function setLockedUntil(?\DateTimeInterface $lockedUntil): self { $this->lockedUntil = $lockedUntil; return $this; }
    public function getLockedBy(): int { return $this->lockedBy; }
    public function setLockedBy(int $lockedBy): self { $this->lockedBy = $lockedBy; return $this; }
    public function isActive(): bool { return $this->isActive; }
    public function setIsActive(bool $isActive): self { $this->isActive = $isActive; return $this; }

    public function isPermanent(): bool { return $this->lockedUntil === null; }
}
